### ISSUE TITLE
[rllib] Error using Tuple action space

### ISSUE THREAD

**Issue by waldroje** [ray-project/ray#13503]
### What is the problem?
I have a custom env where mapping a two or three dimensional choice to a single discrete action space would be cumbersome, and potentially huge, thus trying to use Tuple space.  I have a script that reproduces the error in both frameworks.  The tuple space tested is a (Discrete(9), Box(0,1))... each framework throws it's own error, and it seems to take `tf `longer, but I believe they are a result of the same issue, that somehow the inputs into the softmax are so small, they result in nan output... which presumably means the probability of one or more actions has basically gone to zero.... vanishing/exploding gradient? My actual env has some action choices that should be conditional on the state of the env, and thus can be unlikely choices...  which I have sometimes managed via masking, or simply disregard the action in the env... the errors have occured in both cases.  The error also has occurred using my own custom model as well as the default, which I'm using in the script.  I have reproduced the error in a hacky way by making one of the discrete action choices always get a negative reward... in my actual env, things can run ok for 50-100 iterations... and then fails... any suggestions on how to manage this? 

I'm attaching the script, and error output from both frameworks.

*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray 1.1.0, Python 3.7, TF 2.2, RHEL

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x ] I have verified my script runs in a clean environment and reproduces the issue.
- [x ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).
[error_tf.txt](https://github.com/ray-project/ray/files/5824396/error_tf.txt)
[error_torch.txt](https://github.com/ray-project/ray/files/5824397/error_torch.txt)
[tuple_action.py.gz](https://github.com/ray-project/ray/files/5824398/tuple_action.py.gz)
